server/http/api_v2.go

```diff
@@ -29,6 +29,7 @@ import (
 
 	"github.com/cayleygraph/cayley/clog"
 	"github.com/cayleygraph/cayley/graph"
+	"github.com/cayleygraph/cayley/internal/gephi"
```

Comment:
This is a strange dependency for this package. It would be better if you copy the implementation of `valuesFromString` from `gephi` package instead.

The `MatchIterator` is also not a problem. I would say that it's worth rewriting it differently, so it can take advantage of more query optimizations:
```
var q shape.Quads
// same for other 3 directions
if len(subjects) != 0 {
  q = append(q, shape.QuadFilter{Dir: quad.Subject, Nodes: shape.Lookup(subjects)})
}
it := shape.BuildIterator(q, qs)
```